File: string_problems/MinimizeStringLength.js

// Problem link - https://leetcode.com/problems/minimize-string-length/description/?envType=problem-list-v2&envId=string

let minimizedStringLength = function(s) {
    // step 1 - the minimal length is the number of distinct characters
    return new Set(s).size;
};

// const s = "aaabc";

// const s = "cbbd";

const s = "baadccab";
console.log(minimizedStringLength(s));

File: string_problems/MinimizeStringLength.java

// Problem link - https://leetcode.com/problems/minimize-string-length/description/

package string_problems;

import java.util.*;

public class MinimizeStringLength {
    public static int minimizedStringLength(String s) {
        // step 1 - the minimal length is the number of distinct characters
        Set<Character> uniqueChars = new HashSet<>();
        for(char ch : s.toCharArray()){
            uniqueChars.add(ch);
        }
        return uniqueChars.size();
    }

    public static void main(String[] args) {
        // String s = "aaabc";

        // String s = "cbbd";

        String s = "baadccab";
        System.out.println(minimizedStringLength(s));
    }
}